crates/uv/src/commands/tool/run.rs

```diff
@@ -104,6 +105,26 @@ pub(crate) async fn run(
         return Err(anyhow::anyhow!("Tool command could not be parsed as UTF-8 string. Use `--from` to specify the package name."));
     };
 
+    let possible_path = Path::new(target);
+    if possible_path
+        .extension()
+        .is_some_and(|ext| ext.eq_ignore_ascii_case("py"))
+    {
+        if possible_path.try_exists()? {
```

Comment:
Should we error if we can't read the path? (earnestly not sure here)

Reply:
It's a good question. We could treat that as the "doesn't exist case" if we'd prefer.